frontend/lib/resolution/Resolver.cpp

```diff
@@ -2399,7 +2405,23 @@ QualifiedType Resolver::typeForId(const ID& id, bool localGenericToUnknown) {
     ct = inCompositeType;
   } else {
     if (auto rt = methodReceiverType().type()) {
-      if (auto comprt = rt->getCompositeType()) {
+      // get the new class type if the receiver is a class
+      auto nct = rt->toClassType();
+      // get the manager record using ClassType method managerRecordType()
+      if (nct && nct->managerRecordType(context) &&
+          nct->managerRecordType(context)->id() == parentId) {
+          ct = nct->managerRecordType(context);
+          auto fieldName = parsing::fieldIdToName(context, id);
+          if (fieldName == "chpl_t") {
+            return QualifiedType(QualifiedType::TYPE,
+                                 nct->withDecorator(context,
+                                                    nct->decorator().toBorrowed()));
+          } else if (fieldName == "chpl_p") {
+             return QualifiedType(QualifiedType::VAR,
+                                  nct->withDecorator(context,
+                                                     nct->decorator().toBorrowed()));
```

Comment:
The only thing that differs between the two cases here is the intent.
```suggestion
          CHPL_ASSERT(fieldName == "chpl_t" || fieldName == "chpl_p");
          auto intent = fieldName == "chpl_t" ? QualifiedType::TYPE : QualifiedType::VAR;
          auto borrowed = nct->withDecorator(context, nct->decorator().toBorrowed());
          return QualifiedType(intent, borrowed);
```